tests/test.bal

```diff
@@ -0,0 +1,304 @@
+        log:print(response?.id.toString());
+    } else {
+        test:assertFail(response.message());
+        log:printError(response.message());
+    }
+}
+
+#########################
+# Create Metadata file
+# ######################
+
+@test:Config {
+    dependsOn: [testCreateFolder]
+}
+function testCreateFile() {
+    log:print("Gdrive Client -> testCreateFile()");
+    CreateFileOptional optionalsCreateFile = {
+        ignoreDefaultVisibility : false
+    };
+    File payload_create_file = {
+        mimeType : "application/vnd.google-apps.document",
+        name : fileName
+        //parents : [parentFolder]
+    };
+    File|error response = driveClient->createMetaDataFile(optionalsCreateFile, payload_create_file);
+    //Assertions
+    if(response is File){
+        test:assertNotEquals(response?.id, "", msg = "Expect File id");
+        log:print(response?.id.toString());
+    } else {
+        test:assertFail(response.message());
+        log:printError(response.message());
+    }
+    //Set variable fileId
+    fileId = <@untainted> getIdFromFileResponse(response);
+}
+
+##############################
+# Create Folder with Metadata
+# ############################
+
+@test:Config {}
+function testCreateFolder() {
+    log:print("Gdrive Client -> testCreateFolder()");
+    CreateFileOptional optionalsCreateFolder = {
+        ignoreDefaultVisibility : false
+    };
+    File payloadCreateFolder = {
+        mimeType : "application/vnd.google-apps.folder",
+        name : folderName
+    };
+    File|error response = driveClient->createMetaDataFile(optionalsCreateFolder, payloadCreateFolder);
+    //Assertions
+    if(response is File){
+        test:assertNotEquals(response?.id, "", msg = "Expect File id");
+        log:print(response?.id.toString());
+    } else {
+        test:assertFail(response.message());
+        log:printError(response.message());
+    }
+    parentFolder = <@untainted> getIdFromFileResponse(response);
+}
+
+###################
+# Get files
+# #################
+
+@test:Config {}
+function testGetFiles() {
+    log:print("Gdrive Client -> testGetFiles()");
+    ListFilesOptional optionalSearch = {
+        pageSize : 3
+    };
+    stream<File>|error response = driveClient->getFiles(optionalSearch);
+    if (response is stream<File>){
+        error? e = response.forEach(isolated function (File response) {
+            test:assertNotEquals(response?.id, "", msg = "Expect File id");
+            log:print(response?.id.toString());
+        });
+    } else {
+        test:assertFail(response.message());
+        log:printError(response.message());
+    }
+}
+
+##############
+# Upload File
+# ############
+
+@test:Config {
+    dependsOn: [testCreateFolder]
+}
+function testNewUpload() {
+    log:print("Gdrive Client -> testNewUpload()");
+    UpdateFileMetadataOptional optionals_ = {
+        addParents : parentFolder //Parent folderID
+    };
+    File payload_ = {
```

Comment:
remove underscore. Check all the places and use proper varaible names

Reply:
Did the changes. Thank you @LakshanSS